management/backup.py

```diff
@@ -473,10 +473,91 @@ def list_target_files(config):
 			raise ValueError(e.reason)
 
 		return [(key.name[len(path):], key.size) for key in bucket.list(prefix=path)]
-
+	elif target.scheme == "b2":
+		api = b2_perform_authentication(target.username, target.password)
+		if api == None:
+			raise ValueError("B2 authentication failed for unknown reason.")
+		(auth_token, api_url) = api
+
+		bucket_id = b2_get_bucket_id(target.username, api_url, auth_token, target.hostname)
+		return b2_list_files(target.username, api_url, auth_token, bucket_id, target.path)
 	else:
 		raise ValueError(config["target"])
 
+def b2_perform_authentication(acc_id, app_key):
+	# Perform Authentication with B2 Storage. Returns a tuple
+	# containing the auth token and the API endpoint to use.
+	from urllib.request import Request, urlopen
+	from urllib.error import HTTPError
+	import base64
+	import json
+
+	# Formulate the authtoken according to B2 Storage API Spec
+	auth_token = "Basic" + base64.b64encode((acc_id + ":" + app_key).encode('utf-8')).decode('utf-8')
```

Comment:
This looks fine. It's doing everything the docs say it should.
- The B2 account id and B2 application key for the account are combined into a string in the format "accountId:applicationKey".
- The combined string is Base64 encoded.
- "Basic " is put before the encoded string.